// monaco-graphql/esm/graphqlMode.js
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
define(["require", "exports", "./workerManager", "monaco-editor/esm/vs/basic-languages/graphql/graphql", "./languageFeatures"], function (require, exports, workerManager_1, graphql_1, languageFeatures) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    languageFeatures = __importStar(languageFeatures);
    function setupMode(defaults) {
        const disposables = [];
        const providers = [];
        const client = new workerManager_1.WorkerManager(defaults);
        const { languageId } = defaults;
        disposables.push(client);
        const worker = (...uris) => {
            try {
                return client.getLanguageServiceWorker(...uris);
            }
            catch (err) {
                throw Error('Error fetching graphql language service worker');
            }
        };
        defaults.setWorker(worker);
        monaco.languages.setLanguageConfiguration(languageId, exports.richLanguageConfig);
        monaco.languages.setMonarchTokensProvider(languageId, graphql_1.language);
        function registerFormattingProvider() {
            const { modeConfiguration } = defaults;
            if (modeConfiguration.documentFormattingEdits) {
                providers.push(monaco.languages.registerDocumentFormattingEditProvider(defaults.languageId, new languageFeatures.DocumentFormattingAdapter(worker)));
            }
        }
        function registerProviders() {
            const { modeConfiguration } = defaults;
            disposeAll(providers);
            if (modeConfiguration.completionItems) {
                providers.push(monaco.languages.registerCompletionItemProvider(defaults.languageId, new languageFeatures.CompletionAdapter(worker)));
            }
            if (modeConfiguration.diagnostics) {
                providers.push(new languageFeatures.DiagnosticsAdapter(defaults, worker));
            }
            if (modeConfiguration.hovers) {
                providers.push(monaco.languages.registerHoverProvider(defaults.languageId, new languageFeatures.HoverAdapter(worker)));
            }
            registerFormattingProvider();
        }
        registerProviders();
        let { modeConfiguration, schemaConfig, formattingOptions } = defaults;
        defaults.onDidChange(newDefaults => {
            if (newDefaults.modeConfiguration !== modeConfiguration) {
                modeConfiguration = newDefaults.modeConfiguration;
                registerProviders();
            }
            if (newDefaults.schemaConfig !== schemaConfig) {
                schemaConfig = newDefaults.schemaConfig;
                registerProviders();
            }
            if (newDefaults.formattingOptions !== formattingOptions) {
                formattingOptions = newDefaults.formattingOptions;
                registerFormattingProvider();
            }
        });
        disposables.push(asDisposable(providers));
        return asDisposable(disposables);
    }
    exports.setupMode = setupMode;
    function asDisposable(disposables) {
        return { dispose: () => disposeAll(disposables) };
    }
    function disposeAll(disposables) {
        var _a;
        while (disposables.length) {
            (_a = disposables.pop()) === null || _a === void 0 ? void 0 : _a.dispose();
        }
    }
    exports.richLanguageConfig = {
        comments: {
            lineComment: '#',
        },
        brackets: [
            ['{', '}'],
            ['[', ']'],
            ['(', ')'],
        ],
        autoClosingPairs: [
            { open: '{', close: '}' },
            { open: '[', close: ']' },
            { open: '(', close: ')' },
            { open: '"""', close: '"""', notIn: ['string', 'comment'] },
            { open: '"', close: '"', notIn: ['string', 'comment'] },
        ],
        surroundingPairs: [
            { open: '{', close: '}' },
            { open: '[', close: ']' },
            { open: '(', close: ')' },
            { open: '"""', close: '"""' },
            { open: '"', close: '"' },
        ],
        folding: {
            offSide: true,
        },
    };
});
